Documentation/network/egress-gateway.rst

```diff
@@ -60,6 +60,9 @@ Additionally, the enablement of the egress gateway feature requires that both
 BPF masquerading and the kube-proxy replacement are enabled, which may not be
 possible in all environments (due to, e.g., incompatible kernel versions).
 
+When pods are selected by a DNS policy **and** an Egress gateway policy, the
+DNS proxy needs to operate in transparent mode (``--dnsproxy-enable-transparent-mode=true``).
```

Comment:
Is the dnsproxy transparent mode really needed?

The DNS traffic flows from the Pod to core-dns or node-dns, right? Are there any cases where we need to encap and redirect packets to an external DNS to the GW?

Reply:
@ysksuzuki of course, you're right - the DNS traffic should stay in-cluster, so for EGW it doesn't matter whether we see the original source IP. I must have gotten mixed up with the changes we needed for envoy. Thank you!